tools/bazel.bzl

```diff
@@ -0,0 +1,35 @@
+# Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@bazel_skylib//lib:versions.bzl", "versions")
+
+def require_version(required):
+    """Require the running bazel be a certain version.
+
+    Args:
+      required: the exact version string required
+
+    Returns:
+      None: when the current version matches the required version
+      Fails with a message otherwise
+    """
+
+    current = versions.get()
+    if current != required:
+        fail(
```

Comment:
Is there a way we could make this a warning? I think it should generally be okay if users use a different version, but we can only guarantee 7.0 will work. WDYT?